src/util/dsp/levinson.cpp

```diff
@@ -0,0 +1,103 @@
+//**************************************************************************
+//
+//	$file: main.c $
+//	$author: Martin Fouilleul $
+//	$date: 28/11/2018 $
+//	$description: a simple Toeplitz system solver$
+//
+//**************************************************************************
+
+#include<string.h> // memset()
+
+namespace otto::dsp
+{
+//--------------------------------------------------------------------------
+//
+//	SolveToeplitz() is a (somewhat naive) implementation of the Levinson-Durbin algorithm for solving the Topelitz system Mx = y
+//
+//	arguments :
+//		- size : the size of the Toeplitz matrix and of vectors x and y
+//		- toeplitz : a buffer containing the first column of the matrix from bottom to top, then the first line from left to right.
+//			     eg. if the matrix is :
+//
+//			     [T0 T1   ...  Tn-1]
+//			     [T-1 T0  ...  Tn-2]
+//			     [...              ]
+//			     [T-n+1   ...    T0]
+//
+//			     you should pass the vector [T-n+1 T-n+2 ... T0 T1 ... Tn-1] of length (size+size-1)
```

Comment:
> length `(size+size-1)`

In the case of the LPC, length `size` would suffice, as the correlation matrix is symmetrical.
I would rather have this generic version, though.

A compromise between genericity and performance would be to check if the Toeplitz matrix description vector is of size `size` or `2*size-1`, e.g.
```cpp
if( toeplitz.size() == y.size() ){
    // Symmetrical Matrix
} else{
    // Regular Toeplitz
}
```

Reply:
That seems right, but I wonder what estimation error it yelds compared the non symmetrized version ? 
If we opt for the symmetric matrix, the simplification is quite huge because we don't need to compute a backward and forward vector, as they become reversals of each other.

Reply:
I would definitely optimize and simplify the algorithm for the LPC engine, as the gains are probably worth it.

I'd say let's go for the symmetric matrix. However, this change should be reflected in the comments, to specifically say this is no longer the generic Levinson-Durbin.

Or make two functions `solveToeplitz()` and `solveSymmetricalToeplitz()` ? We'd then use the symmetrical version for the LPC.

Then again, if we never use the non-symmetrical version, is it even worth introucing zombie code ? 🤔 Probably not. But I like the idea of having a generic LD around somewhere.

Reply:
Yes I agree that we should optimize for our specific use case and rename the function accordingly !
And imo it's not useful to keep the generic one if it's not used. But in any case that code won't be lost, so if we need it later it will still be available.
Also, the more I think about it the more I believe there's no significant difference in the estimation error between the symmetrical vs. non-symmetrical case (although I have no formal proof...)

Reply:
It occurred to me that in our specific use case, not only the correlation matrix is symmetrical, but we also know that the right-hand vector **y** has only one non-zero coordinate (the first, which is equal to one). So we could even simplify a bit further ?